lib/automaticupgrades/channel.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package automaticupgrades
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/maintenance"
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/version"
+)
+
+// Channels is a map of Channel objects.
+type Channels map[string]*Channel
+
+func (c Channels) CheckAndSetDefaults() error {
+	var errs []error
+	var err error
+	for name, channel := range c {
+		err = trace.Wrap(channel.CheckAndSetDefaults(), "failed to create channel %s", name)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return trace.NewAggregate(errs...)
+}
+
+// Channel describes an automatic update channel configuration.
+// It can be configured to serve a static version, or forward version requests
+// to an upstream version server. Forwarded results are cached for 1 minute.
+type Channel struct {
```

Comment:
This is a good idea but I think we would be conflicting with some goals from security: they are trying to hide the minor/bugfix versions from the ping endpoint to make it harder for an attacker to identify if the teleport instance is out-of-date and vulnerable.

If people from cloud and security are OK I'll gladly implement that, I think it would solve the needs of 99% of self-hosted users.